src/app/routes/getIndexData/index.js

```diff
@@ -0,0 +1,50 @@
+import 'isomorphic-fetch';
+import nodeLogger from '../../helpers/logger.node';
+import getBaseUrl from '../helpers/getBaseUrl';
+
+const logger = nodeLogger(__filename);
+
+const upstreamStatusCodesToPropagate = [200, 404];
+
+const getIndexData = async ({ match }) => {
+  const { service } = match.params;
+  let data;
+  let status;
+  let baseUrl = process.env.SIMORGH_BASE_URL;
+
+  if (
+    typeof window !== 'undefined' &&
+    window.location &&
+    window.location.origin
+  ) {
+    baseUrl = getBaseUrl(window.location.origin);
+  }
+
+  const url = `${baseUrl}/${service}/index/front_page.json`;
+
+  try {
```

Comment:
I know its POC but we should share allllll the thingssss

Reply:
This is why originally I was passing a pageType props to `getInitialData` and was constructing the url based off of that pageType value e.e. `article` or `index`.

However you mentioned the preProcessor  and I am not sure how that overlaps with `getInitial` data yet.

With this setup I personally do not see the point in have `getArticleData` and `getIndexData` when 98% of the code is the same